include/triton/backend/backend_memory.h

```diff
@@ -111,9 +120,10 @@ class BackendMemory {
  private:
   BackendMemory(
       TRITONBACKEND_MemoryManager* manager, const AllocationType alloctype,
-      const int64_t memtype_id, char* buffer, const size_t byte_size)
+      const int64_t memtype_id, char* buffer, const size_t byte_size,
+      const bool owns_buffer)
       : manager_(manager), alloctype_(alloctype), memtype_id_(memtype_id),
-        buffer_(buffer), byte_size_(byte_size)
+        buffer_(buffer), byte_size_(byte_size), owns_buffer_(owns_buffer)
```

Comment:
Can you make this constructor to default the owns buffer to true when not specified?